eloquentjs/4_3_list.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Converts an array  to a list data structure
+ * @param {*} inputArray 
+ * @param {*} list 
+ * @returns 
+ */
+function arrayToList([...inputArray], list = { value: null, rest: null }) { 
```

Comment:
Doing a shallow copy of `inputArray` for each element in it is going to be very slow for not much reason. If you add some simple index tracking logic you can avoid the array copies. I think it would be good practice to try doing this both recursively and iteratively without the copy.